frontend/src/metabase/query_builder/actions/core/updateQuestion.ts

```diff
@@ -144,6 +144,13 @@ export const updateQuestion = (
       }
     }
 
+    // This scenario happens because the DatasetQueryEditor converts the dataset/model question into a normal question
+    // so that its query is shown properly in the notebook editor. Various child components of the notebook editor have access to
+    // this `updateQuestion` action, so they end up triggering the action with the altered question.
+    if (queryBuilderMode === "dataset" && !newQuestion.isDataset()) {
+      newQuestion = newQuestion.setDataset(true);
+    }
+
```

Comment:
1. perhaps! may need to introduce some new concepts around what we store as the primary question in the QB since so much is dependent on the state of the primary `state.qb.question` question.
2. done